backend/app/adapter/gqlapi/resolver/authmutation_test.go

```diff
@@ -0,0 +1,235 @@
+		expectedShortLink  *URL
+		shouldError        bool
+	}{
+		{
+			name: "empty update returns empty response",
+			args: &UpdateURLArgs{
+				OldAlias: "mySimpleAlias",
+				URL:      URLInput{},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: nil,
+			shouldError:       false,
+		},
+		{
+			name: "update short link alias",
+			args: &UpdateURLArgs{
+				OldAlias: "mySimpleAlias",
+				URL: URLInput{
+					CustomAlias: &newAlias,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    newAlias,
+					LongLink: "https://www.google.com/",
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update only long link",
+			args: &UpdateURLArgs{
+				OldAlias: "mySimpleAlias",
+				URL: URLInput{
+					OriginalURL: &newLongLink,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update both alias and long link",
+			args: &UpdateURLArgs{
+				OldAlias: "mySimpleAlias",
+				URL: URLInput{
+					CustomAlias: &newAlias,
+					OriginalURL: &newLongLink,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    newAlias,
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			blockedHash := map[string]bool{
+				"http://malware.wicar.org/data/ms14_064_ole_not_xp.html": false,
```

Comment:
Let's add a test case for updating `long_link` to a malicious one.

Reply:
Definitely will add this test case.